FormatJavaCucumber.js

```diff
@@ -1,113 +1,140 @@
+                "content_type": attachment.mime_type,
+                data: attachment.data
+            }
+};
+
+// Iterates over each step and grabs all attachments and returns a flat array containing the attachment information
+const getStepAttachments = testCase => testCase.steps.filter(hasEmbeddings).map(attachmentsWithStepNameInjected)
+    .reduce(flattenArray, []).map(attachmentInformation);
+
+// Iterates over each hook and grabs all attachments and returns a flat array containing the attachment information
+const getHookAttachments = testCase => ("after" in testCase) ? testCase.after.filter(hasEmbeddings).map(attachmentsWithStepNameInjected)
+    .reduce(flattenArray, []).map(attachmentInformation) : [];
+
+// Grabs all attachments from the hooks and steps and combines them into a single flat array of attachment information
+const getAllAttachments = testCase => getStepAttachments(testCase).concat(getHookAttachments(testCase));
+
+// Enum object to handle possible statuses within cucumber json output and in qTest Manager
+const Status = {
+    PASSED: "passed",
+    FAILED: "failed",
+    SKIPPED: "skipped",
+    UNDEFINED: "undefined",
+    PENDING: "pending",
+    BLOCKED: "blocked",
+}
+
+// Calculates the overall testcase status based on the result of the passed in step, storing the result in the accumulator
+const testCaseStatus = (acc, step) => (Status.PASSED === acc) ? step.result.status: acc;
+
+// Gets the testcase status based on the result of each step
+const getTCStatus = testCase => testCase.steps.reduce(testCaseStatus, Status.PASSED);
+
+// Returns an  actual result based on step.result.status
+// A step is skipped when a previous step, background step or before hook fails
+// A step is undefined when it exists in a feature but no definition is found
+// A step is pending when it exists in a feature file, has a defition, but explicitly throws a PendingException
+const getActualResult = step => {
+    return {
+        [Status.PASSED]: step.name,
+        [Status.FAILED]: step.result.error_message,
+        [Status.SKIPPED]: "This step has been skipped due to a previous failure",
+        [Status.UNDEFINED]: "This step does not match any step definitions",
+        [Status.PENDING]: "This step is marked as pending"
+    }[step.result.status];
+}
+
+// Generates a step log object for injection into a test log
+const testStepLogs = testCase => testCase.steps.map((step, index) => {
+    return {
+        order: index,
+        description: `${step.keyword}`,
+        expected_result: step.name,
+        actual_result: getActualResult(step),
+        status: step.result.status
+    };
+});
+
+// Injects the feature name and URI into the test case object, relying on the context of 'this' to be set to a test case object 
+// e.g. an arrow function cannot be used
+const injectFeatureNameAndUri = function(testCase) {
+    testCase.feature_uri = this.uri;
+    testCase.feature_name = this.name;
+    return testCase;
```

Comment:
I'm confused, context is a test case, and we pass in a test case? Are they the same?

Reply:
Should have been that the context of this was a feature object.  I've corrected this thanks.